native-schema-registry/python/PyGsrSerDe/AvroSerDe.py

```diff
@@ -0,0 +1,74 @@
+import fastavro
+from io import BytesIO
+
+
+class AvroSerializer:
+    """A serializer for serializing AVRO records into bytes."""
+
+    # NOTE(allkliu): Parsing the schema manually is not necessary, as the fastavro.writer
+    # will parse it automatically. However, parsing and saving the schema
+    # saves time if calling serialize repeatedly.
+    def parse_schema(self, schema: dict):
+        """Parse and return the schema.
+
+        Args:
+            schema -- pass in a schema to parse
+
+        Returns:
+            A parsed schema.
+        """
+        return fastavro.parse_schema(schema)
+
+    def serialize(self, data: dict, schema: dict):
+        """Serialize a record and return the serialized bytes.
```

Comment:
It is the case that serialize method will serialize a list of records right?

By the way, I assume it is the case that fastavro  library excepts a list of avro records as input, what will happen if customer wants to serialize one record instead of a list of records, is it the case that they must place that single record in a list data structure?

Reply:
Yes, I mistakenly wrote in the docstrings that it should be a dictionary, but I just double checked and fastavro.write expects an iterable as data. It returns 'ValueError: "records" argument should be an iterable, not dict' otherwise. It seems like it must be in a list even if the customer wants to serialize on record.